code/sorting/src/bucket_sort/README.md

```diff
@@ -1,4 +1,36 @@
-# cosmos
-Your personal library of every algorithm and data structure code that you will ever encounter.
+# Bucket Sort
+Bucket sort is a sorting algorithm, that works by distributing the elements of an array into a number of buckets. Then, we apply some sorting algorithm (**insertion sort** for best optimisation ) to sort elements in each bucket, & finally, take the elements out of buckets in order & put them back to original array to get sorted result.
 
-A large scale collaboration of [OpenGenus](https://github.com/opengenus)
+## Example
+
+![fig1](https://upload.wikimedia.org/wikipedia/commons/6/61/Bucket_sort_1.svg)
+
+Elements are first distributed among bins/buckets
+
+![fig2](https://upload.wikimedia.org/wikipedia/commons/e/e3/Bucket_sort_2.svg)
+
+Then, elements are sorted within each bucket.
+
+## Algorithm
+Bucket sort is mainly useful when input is uniformly distributed over a range.
+
+If input is an n element array arr[i] with all elements lying in the range: [0.0, 1.0], & uniformly distributed in the same range, the algorithm is:
```

Comment:
& -> and

Reply:
Will take care of this from next time. 👍